### PR TITLE
Add Breaking changes section to SG cookbook

### PR DESCRIPTION
**Pull request by chsienki**

### CODE REVIEW

File: docs/features/source-generators.cookbook.md [hunk 1]
```diff
@@ -857,6 +857,32 @@ specialized exactly to what was written in the user class.
...
 TODO:
 
+## Breaking Changes:
+
+Between preview and release the following breaking changes were introduced:
+
+Rename `SourceGeneratorContext` to `GeneratorExecutionContext`
+Rename `IntializationContext` to `GeneratorInitializationContext`
+
+This affects users authored generators, as it means the base interface has changed to:
```
Comment (cston) [inline docs/features/source-generators.cookbook.md:867]: Perhaps "user-authored".

File: docs/features/source-generators.cookbook.md [hunk 2]
```diff
@@ -857,6 +857,32 @@ specialized exactly to what was written in the user class.
 
 TODO:
 
+## Breaking Changes:
+
+Between preview and release the following breaking changes were introduced:
+
+Rename `SourceGeneratorContext` to `GeneratorExecutionContext`
+Rename `IntializationContext` to `GeneratorInitializationContext`
```
Comment (Youssef1313) [inline docs/features/source-generators.cookbook.md:865]:
```suggestion
- Rename `SourceGeneratorContext` to `GeneratorExecutionContext`
- Rename `IntializationContext` to `GeneratorInitializationContext`
```

File: docs/features/source-generators.cookbook.md [hunk 3]
```diff
@@ -857,6 +857,32 @@ specialized exactly to what was written in the user class.
...
+
+This affects users authored generators, as it means the base interface has changed to:
+
+```csharp
+    public interface ISourceGenerator
+    {
+        void Initialize(GeneratorInitializationContext context);
+        void Execute(GeneratorExecutionContext context);
+    }
```
Comment (Youssef1313) [inline docs/features/source-generators.cookbook.md:874]:
```suggestion
public interface ISourceGenerator
{
    void Initialize(GeneratorInitializationContext context);
    void Execute(GeneratorExecutionContext context);
}
```

File: docs/features/source-generators.cookbook.md [hunk 4]
```diff
@@ -857,6 +857,32 @@ specialized exactly to what was written in the user class.
...
+```
+
+The required action from the user is to rename the parameter types of the `Initialize` and `Execute` methods to match.
+
+**Rename `RunFullGeneration` to `RunGeneratorsAndUpdateCompilation`**
+**Add `Create()` static methods to `CSharpGeneratorDriver` and obsolete the constructor.**
```
Comment (Youssef1313) [inline docs/features/source-generators.cookbook.md:880]: Before making these list item, they will appear in **the same** line.

```suggestion
- **Rename `RunFullGeneration` to `RunGeneratorsAndUpdateCompilation`**
- **Add `Create()` static methods to `CSharpGeneratorDriver` and obsolete the constructor.**
```

**Pull request merged**